src/sink/tcp.rs

```diff
@@ -85,8 +106,31 @@ impl Tcp {
         }
         Ok(())
     }
+
+    async fn connect(config: &Config) -> Result<Stream> {
+        let stream = TcpStream::connect((config.host.as_str(), config.port)).await?;
+        stream.set_ttl(config.ttl)?;
+        stream.set_nodelay(config.is_no_delay)?;
+        let s: Stream = match config.tls.as_ref() {
```

Comment:
could you add a debug log to each of those branches? That would be helpful when debugging some issues